convert.py

```diff
@@ -80,18 +119,21 @@ def main():
 
                 properties = modelSpec.get('properties', {})
 
+                resource_data = get_static_data(modelSpec)
+                param_names = required - set(resource_data.keys())
```

Comment:
I wonder if we could just remove `apiVersion` and `kind` from the "artificially" required keys [here](https://github.com/geigerzaehler/dhall-kubernetes/blob/1feeeac87a05be391d6cf8d03b589a41e89b03a0/convert.py#L87) instead of doing this subtraction. Is it so that they are _always_ present in every model spec?

Reply:
If we were to remove the `always_required` keys from the `required` set the types would look different for these resources. `kind` would have type `Optional Text` instead of type `Text`. This was the rationale for this approach.

Reply:
Oh right, great 👍